cli/lib/kontena/cli/stacks/yaml/reader.rb

```diff
@@ -147,10 +157,18 @@ def process_variables
         raise RuntimeError, "Variable validation failed: #{variables.errors.inspect}" unless variables.valid?
       end
 
+      # @raise [Liquid::Error]
       def interpolate_liquid(content, vars)
         Liquid::Template.error_mode = :strict
         template = Liquid::Template.parse(content)
-        template.render(vars, strict_variables: true, strict_filters: true)
+
+        # Wrap nil values in LiquidNull to not have Liquid consider them as undefined
+        vars = vars.clone
```

Comment:
I assume the stack variables are always going to be a flat hash... I suppose Liquid does have some support for indexing into hash values, but I'm not sure how it then deals with nil values in there..